lstchain/calib/camera/calibrator.py

```diff
@@ -100,7 +102,9 @@ def __init__(self, subarray, **kwargs):
             )
 
         # declare time calibrator if correction file exist
-        if os.path.exists(self.time_calibration_path):
+        if self.time_calibration_path == '':
+            self.time_corrector = None
+        elif os.path.exists(self.time_calibration_path):
```

Comment:
Hi @jsitarek ,
I think we decided we want always the time calibration file, one could indeed take out the "if self.time_corrector:" in the lines below

Reply:
Hi @FrancaCassol 

for regular processing of the data yes, it is needed and should be used
I added those lines to be able to do tests removing such a correction (I did not find any other way to do it in the current code), because Abelardo was asking how the time resolution would look like without this correction. Since there are still discussions how to correct the time in waveforms that go to ACADA, and we might need to do some more tests for this in the near future I think it makes is useful to have such an option in the code

Reply:
Hi @jsitarek, 
I leave @maxnoe to review this, I thinks he asked such a security in the input trailets, in order not to risk a wrong processing of real data (which indeed demand to have the time calibration file). 
In fact, if you are testing new algorithms, you are anyway touching the code and you can locally modify this trailet.

Reply:
Yes, please don't silently fail on a missing input file. If the user give a time calibration file and it doesn't exist, an error must be raised.

You should use `None` for `no time calibration file given` and then check `if self.time_calibration_path is None`.

Also this is already implemented in the new source. Why do we need this here at all?

Reply:
Hi @maxnoe 

stricktly speaking, the user is *not* giving a path to a file, if the path variable is set to an empty string, but I understand 

this PR started in early November last year as a fix of 1 character in a doc string following an issue reported by @maxnoe 
then it was not merged still in the end of November when by mistake I added to it a separate PR with computing time resolution and adding this possibility to switch off the DRS4 time correction.

Since both the spikes interpolation (which was the original PR) and the DRS4 time calibration (which was the second one) now will go into general ctapipe due to PR #587 I agree with @maxnoe that it is not needed anymore, so I will close the PR

sorry for a waste of time for everybody involved

Reply:
> stricktly speaking, the user is not giving a path to a file, if the path variable is set to an empty string, but I understand

Then test for empty string and not for `exists`.

Reply:
Sorry, I do not understand your point. This is exactly what the change was. The old code was testing if the file exist and if so, switching on the correction, otherwise raises an error. The change that I proposed first checks if the string is empty, if so, it switches off  the correction, then it proceeds with the old behaviour (i.e. raises an error if path is wrong, or uses the correction if it is ok).

Reply:
That behaviour you are describing is correct.